### PR TITLE
TINKERPOP-1443 API Checker

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-1443

This is basically the continuation to #494 - didn't really change anything though @dkuppitz did get docker working. Travis seems ok with the build as well. No one seemed to have a problem with requiring maven 3.2.5 or better for our build (see the linked DISCUSS thread in JIRA), so keep in mind that if you run an earlier version it will cause trouble (only on the master branch obviously). 

I guess we will need a release of 3.3.0 before we will have any real use out of this as breaking changes are allowed between our last release on 3.2.x and our future 3.3.0. 

All tests pass with `docker/build.sh -t -n -i`

VOTE +1

### CODE REVIEW

**Comment by dkuppitz** [comment]
VOTE: +1

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,18 @@ jdk:
   - oraclejdk8
 sudo: required
 dist: trusty
+cache:
+  directories:
+    - $HOME/mvn-home
+
+install:
+  # install maven 3.3.9 if it is not in cache already
```
Comment (spmallette) [inline .travis.yml]: Pushed fixes for both of those. Note that 3.2.5 is the actual minimum maven version required.

Comment (pluradj) [inline .travis.yml]: then shouldn't Travis use 3.2.5?

Comment (spmallette) [inline .travis.yml]: I'm not sure I see why it matters. The requirement is at least 3.2.5 and 3.3.9 meets that requirement. I suppose we could pin travis to the minimum version but is there any particular reason to do so?

Comment (pluradj) [inline .travis.yml]: right, consistency and built-in testing for the stated minimum Maven level. probably no big deal either way.

**Comment by pluradj** [comment]
VOTE: +1

**Comment by spmallette** [comment]
@pluradj i changed it to 3.2.5 for travis - travis looks happy with the build so i'll prepare to merge today.

**Pull request merged**